### PR TITLE
feat: basic implementation of SetAddElements and SetRemoveElements

### PR DESCRIPTION
**Pull request by virratanasangpunth**
There is a few small TODOs left that can be fast-followed after this
PR is merged.

Integ tests are not added yet, that will be done in the next PR.

### CODE REVIEW

File: src/internal/cache-client.ts [hunk 1]
```diff
@@ -196,6 +198,107 @@ export class CacheClient {
...
+    const metadata = this.createMetadata(cacheName);
+    return await new Promise(resolve => {
+      this.clientWrapper.getClient().SetUnion(
+        request,
+        metadata,
+        {
+          interceptors: this.interceptors,
+        },
+        (err, resp) => {
+          if (err) {
```
Comment (cprice404) [inline src/internal/cache-client.ts:240]: nit - lint warnings above can be avoided by simply omitting `resp` from your argument list here.  (I think this is an annoying language design choice and I think your way is more clear though so if you wanted to add the one-line eslint-ignore comment on that arg list that would be fine too.

File: src/messages/responses/cache-set-add-elements.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import {ResponseBase} from './response-base';
+import {SdkError} from '../../errors/errors';
+import {applyMixins, ErrorBody} from '../../errors/error-utils';
+import {TextDecoder} from 'util';
+
```
Comment (cprice404) [inline src/messages/responses/cache-set-add-elements.ts:5]: legit warning

**Review by cprice404 — APPROVED** [review]
couple nits re: lint warnings but overall lgtm, and i don't care if you fast-follow those lint warnings personally.  but i guess we need to align on merge order convo this morning.

File: src/internal/cache-client.ts (hunk 1 above)
Comment (virratanasangpunth) [reply src/internal/cache-client.ts:240]: kk gonna tackle these as fast-follows: https://github.com/momentohq/client-sdk-javascript/issues/178

**Pull request merged**